docs/tour.mdx

```diff
@@ -1022,20 +1812,52 @@ const checkout = async (
 };
 ```
 
+</TabItem>
+<TabItem value="java" label="Java">
+
+```java
+import java.util.UUID;
+
+public BoolValue checkout(RestateContext ctx, CheckoutFlowRequest request) throws TerminalException {
+    // highlight-next-line
+    var idempotencyKey = ctx.sideEffect(CoreSerdes.STRING_UTF8, () -> UUID.randomUUID().toString());
+    return BoolValue.of(true);
+}
+```
+
+</TabItem>
+</Tabs>
+
 The highlighted line of code wraps the creation of the UUID in a side effect with a string return type.
 When the `checkout()` function gets re-executed upon replay, Restate injects the stored value.
 
-To experiment, you can add a log line and a sleep to see how the UUID gets replayed after the suspension:
+To experiment, you can print the idempotency key to the console and add a sleep to see how the UUID gets replayed after the suspension:
+
+<Tabs groupId="sdk">
+<TabItem value="ts" label="TypeScript">
 
 ```ts
 console.log("My idempotency key: " + idempotencyKey);
 await ctx.sleep(31000);
 ```
+</TabItem>
+<TabItem value="java" label="Java">
+
+```java
+System.out.println("My idempotency key: " + idempotencyKey);
+ctx.sleep(Duration.ofSeconds(31));
```

Comment:
Here we probably need to sleep for 61 seconds to trigger the suspension. I think it would be best to align the default suspension timeout for both SDKs. Otherwise we will always have something like this slip through the cracks.